### PR TITLE
feat(builder): assemble execution payload bids

### PR DESCRIPTION
**Pull request by krisoshea-eth**
## Motivation

The Builder needs one explicit, tested boundary for turning exact Engine payload material into the fork-correct `ExecutionPayloadBid` before pricing, signing, publication, and slot orchestration are wired together.

Nico's `nflaig/builder` proof of concept performs this mapping inside `SlotBidder`. This PR extracts only the message assembly so the protocol fields and later-fork behavior can be reviewed independently.

## Changes

- Assemble all common bid fields from an exact `BuiltPayload` plus caller-owned proposal context.
- Force `executionPayment` to zero for the p2p trustless bid path.
- Hash execution requests with the matching fork type.
- Check at runtime that the payload's fork matches the bid's fork.
- Require Heze `inclusionListBits` explicitly instead of silently dropping the fork field.
- Reject runtime payload-fork mismatches and invalid or unsafe numeric bid values with structured errors. Reject slot mismatches and payloads whose block hash equals their parent hash before assembling a bid.

## Scope

This PR does not compute bid value, check balance or coverability, retain payload material, select proposer preferences, sign or publish bids, schedule slot work, or wire the Builder runtime.

#9958 has merged, and `BuiltPayload` is the plain type after its review. This branch is merged forward to `unstable`, so the diff is the two bid-assembly files.

Tracking: [LOD-62](https://linear.app/kriso/issue/LOD-62/bid-assembly-01-assemble-fork-correct-execution-payload-bids)

Reference: `nflaig/builder` at `99fd8fa9ad`

Historical validation base: `f22c5ce63e`

Validated head: `64630dc`

## Testing

At `64630dc`:

- 10 focused bid-assembly tests, including coherent slot fixtures and the same-hash rejection from #9972
- type-check and Biome on the changed files

Previously validated on 2026-09-07 at `a7f5146fe2` with Node 24.13.0: Builder package type-check, Biome on the changed files, Builder build and module import check, `git diff --check`.

These are local results, not a claim about GitHub CI or a live BN/EL run.

## Review plan

Treat this and #9979 as one logical bid assembly/publication path. If maintainers prefer fewer service-only PRs, combine the two around their first runtime consumer.

## AI assistance

AI assistance was used during codebase research, implementation, drafting, testing, and review. The submitted code and PR text were reviewed and revised by the author, including technical decisions.

### CODE REVIEW

**Comment by markolazic01** [comment]
lgtm

File: packages/builder/src/services/executionPayloadBid.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    executionPayment: 0n,
+    blobKzgCommitments: blobsBundle.commitments,
+    executionRequestsRoot:
+      input.fork === ForkName.heze
+        ? ssz.heze.ExecutionRequests.hashTreeRoot(executionRequests)
+        : ssz.gloas.ExecutionRequests.hashTreeRoot(executionRequests),
+  };
+
+  if (input.fork === ForkName.heze) {
+    return {...bid, inclusionListBits: input.inclusionListBits};
```
Comment (nazarhussain) [inline packages/builder/src/services/executionPayloadBid.ts]: Its SSZ type is `BitVectorType(INCLUSION_LIST_COMMITTEE_SIZE)`, but it is accepted as a bare `BitArray` and never length-checked. A shorter `bitLen` is silently zero-padded during `serializeToBytes`, producing a signing root over a bid the caller never intended. A longer one throws a `RangeError` deep inside serialization at publish time, far from the cause.

Every other field gets an explicit runtime check, which is what makes this one stand out. Suggest a `bitLen === INCLUSION_LIST_COMMITTEE_SIZE` check with its own error code.

File: packages/builder/src/services/executionPayloadBid.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+type CommonBidInput<F extends ForkName.gloas | ForkName.heze> = {
+  fork: F;
+  slot: Slot;
+  parentBlockRoot: Root;
+  builderIndex: BuilderIndex;
+  feeRecipient: ExecutionAddress;
+  value: number;
+  payload: BuiltPayload;
+};
```
Comment (nazarhussain) [inline packages/builder/src/services/executionPayloadBid.ts]: `EnginePayloadSource.getPayload` returns `BuiltPayload` with `fork: ForkPostGloas`. So better we use the same `ForkPostGloas` here and not hardcode `ForkName.gloas | ForkName.heze`
